Clarify situation when enabling servers (#2623)
neo4j/docs-operations: modules/ROOT/pages/clustering/servers.adoc

[[server-enabled-state]]
=== Enabled state

> When you deploy a cluster for the first time, all servers included in the initial deployment are automatically enabled.

Reviewer: I'm afraid it's more annoying than this. The strict behaviour is:
1. When creating the cluster for the first time, it will wait until there are at least `dbms.cluster.minimum_initial_system_primaries_count` `system` primaries available.
2. At that point, the `system` database / Raft group will be created, and all servers present at that point, from the point of view of the one server that is creating the `system` database, will be added and enabled.
3. Any servers that were not visible to that bootstrapper will still be considered `Free`.

So if you start with a minimum of 3, but start 5 servers, you will get between 3 and 5 enabled servers, depending on timings. You can dodge some of this complexity if you only have `system` primaries, by setting `minimum_initial_system_primaries_count` to the full number of servers. But if you have `system` secondaries? You're exposed to this.

Maybe a friendlier summary might be something like:

```suggestion
When you deploy a cluster for the first time, at least the minimum number of servers included in the initial deployment are automatically enabled.
```

For details, refer to the example on how to xref:clustering/setup/deploy.adoc#cluster-example-configure-a-three-primary-cluster[Configure a cluster with three servers].

If you add a new server after the cluster is already running, the server is added in the `FREE` state.
A server in the `FREE` state needs to be explicitly enabled in order to be considered an active member of the cluster.

To transition a server into the _Enabled_ state, use the command `ENABLE SERVER`:

[source, syntax, role=noheader]
----

> ENABLE SERVER 'serverId' [OPTIONS "{" option: value[,...] "}"]

Author: What should we use here: 'serverId' or 'name'?
The next sentence says that the server's initial name is its ID. But we don't mention its name above.

[NOTE]
====

> When a server is enabled, if no `OPTIONS` are not provided, the default server's values are taken from the settings:

Reviewer: Double negative
```suggestion
When a server is enabled, if no `OPTIONS` are provided, the default server's values are taken from the settings:
```

* xref:configuration/configuration-settings.adoc#config_initial.server.mode_constraint[`initial.server.mode_constraint`]
* xref:configuration/configuration-settings.adoc#config_initial.server.allowed_databases[`initial.server.allowed_databases`]
* xref:configuration/configuration-settings.adoc#config_initial.server.denied_databases[`initial.server.denied_databases`]
* and/or xref:configuration/configuration-settings.adoc#config_initial.server.tags[`initial.server.tags`]

> All these settings are only effective on the first startup of the DBMS.

Reviewer: Because these settings are `initial.server`, not `initial.dbms`, they are only effective when enabling the server.

```suggestion
All these settings are only effective when enabling the relevant server.
```

====

> Another option to enable a server is to set xref:configuration/configuration-settings.adoc#config_initial.dbms.automatically_enable_free_servers[`initial.dbms.automatically_enable_free_servers`] to `true`.

Reviewer: ```suggestion
Another option to enable servers is to have the DBMS automatically enable free servers.
Do this either by setting xref:configuration/configuration-settings.adoc#config_initial.dbms.automatically_enable_free_servers[`initial.dbms.automatically_enable_free_servers`] to `true` before starting the deployment for the first time, or running `CALL dbms.cluster.setAutomaticallyEnableFreeServers(true);`.
```
